### PR TITLE
added critique library and tests

### PR DESCRIPTION
**Pull request by mattjones315**

### CODE REVIEW

File: cassiopeia/critique/compare.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+        all_triplets_correct[depth] /= number_of_trials
+        resolvable_triplets_correct[depth] /= (
+            number_of_trials - number_unresolvable_triplets
+        )
+        
+        if number_unresolvable_triplets == 0:
+            unresolved_triplets_correct[depth] = 1.0
+        else:
+            unresolved_triplets_correct[depth] /= max(1, number_unresolvable_triplets)
```
Comment (richardyz98) [inline cassiopeia/critique/compare.py:116]: Is the max necessary?

File: cassiopeia/critique/compare.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+            all_triplets_correct[depth] += score
+            if is_resolvable:
+                resolvable_triplets_correct[depth] += score
+            else:
+                unresolved_triplets_correct[depth] += score
+
+        all_triplets_correct[depth] /= number_of_trials
+        resolvable_triplets_correct[depth] /= (
+            number_of_trials - number_unresolvable_triplets
```
Comment (richardyz98) [inline cassiopeia/critique/compare.py:110]: Maybe we should also create a case where all of the triplets are unresolvable? In which case we would want to do the 0 denominator check as well

File: cassiopeia/critique/compare.py (hunk 1 above)
Comment (mattjones315) [reply cassiopeia/critique/compare.py:116]: No, good catch! I'll remove it.

File: cassiopeia/solver/solver_utilities.py [hunk 3]
```diff
@@ -165,15 +166,42 @@ def collapse_tree(
...
+
+    collapse_fn = lambda x: (len(x.children) == 1)
+
+    collapsed_tree = tree.copy()
+    to_collapse = [n for n in collapsed_tree.traverse() if collapse_fn(n)]
+
+    for n in to_collapse:
+        n.delete()
+
+    return collapsed_tree
```
Comment (richardyz98) [inline cassiopeia/solver/solver_utilities.py:190]: Are the children of the nodes attached to the parent of the deleted node?

File: test/solver_tests/edge_collapse_test.py [hunk 4]
```diff
@@ -137,6 +138,42 @@ def test_newick_converter(self):
...
+        collapsed_tree = solver_utilities.collapse_unifurcations(tree)
+
+        # make sure all leaves remain
+        self.assertEqual(len(tree), len(collapsed_tree))
+        for n in tree:
+            self.assertIn(n.name, collapsed_tree.get_leaf_names())
+
+        # make sure there are no singletons left
+        for n in collapsed_tree.traverse():
+            self.assertFalse(len(n.children) == 1)
```
Comment (richardyz98) [inline test/solver_tests/edge_collapse_test.py:157]: Maybe we should also test that the final tree is connected, and that the nodes that should be connected are connected?

**Review by richardyz98 — CHANGES_REQUESTED** [review]

File: test/solver_tests/edge_collapse_test.py (hunk 4 above)
Comment (richardyz98) [reply test/solver_tests/edge_collapse_test.py:157]: for example making sure that node 0 is connected to node 3 in the final tree?

File: cassiopeia/solver/solver_utilities.py (hunk 3 above)
Comment (mattjones315) [reply cassiopeia/solver/solver_utilities.py:190]: Yes, this is done whenever you delete a given node: http://etetoolkit.org/docs/latest/reference/reference_tree.html#ete3.TreeNode.delete

File: test/solver_tests/edge_collapse_test.py (hunk 4 above)
Comment (mattjones315) [reply test/solver_tests/edge_collapse_test.py]: Done.

File: cassiopeia/critique/compare.py (hunk 2 above)
Comment (mattjones315) [reply cassiopeia/critique/compare.py]: Done.

**Review by richardyz98 — APPROVED** [review]

**Pull request merged**